cpu/cortexm_common/include/bitband.h

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2017 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_cortexm_common
+ * @{
+ *
+ * @file
+ * @brief       Bit access macros for Cortex-M based CPUs
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ */
+
+#ifndef BITBAND_H
+#define BITBAND_H
+
+#include <stdint.h>
+#include "cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Define BITBAND_FUNCTIONS_PROVIDED 1 if the CPU provides its own
+ * implementations for bit manipulation */
+#if !BITBAND_FUNCTIONS_PROVIDED
+
+#if DOXYGEN
+/** @brief Flag for telling if the CPU has hardware bit band support */
+#define CPU_HAS_BITBAND 1 || 0 (1 for Cortex-M3 and up, 0 for Cortex-M0)
+#endif
+
+#ifndef CPU_HAS_BITBAND
+#if (__CORTEX_M >= 3)
+#define CPU_HAS_BITBAND 1
+#else
+#define CPU_HAS_BITBAND 0
+#endif
+#endif
+
+#if CPU_HAS_BITBAND || DOXYGEN
+/* Cortex-M3 and higher provide a bitband address space for atomically accessing
+ * single bits of peripheral registers, and sometimes for RAM as well */
+/**
+ * @name Bit manipulation functions
+ * @{
+ */
+/* Generic bit band conversion routine */
+/**
+ * @brief Convert bit band region address and bit number to bit band alias address
+ *
+ * @param[in] ptr  base address in non bit banded memory
+ * @param[in] bit  bit number within the word
+ *
+ * @return Address of the bit within the bit band memory region
+ */
+static inline volatile void *bitband_addr(volatile void *ptr, uintptr_t bit)
+{
+    return (volatile void *)((((uintptr_t)ptr) & 0xF0000000ul) + 0x2000000ul +
+        ((((uintptr_t)ptr) & 0xFFFFFul) << 5) + (bit << 2));
+}
+
+/**
+ * @brief Set a single bit in the 32 bit word pointed to by @p ptr
+ *
+ * The effect is the same as for the following snippet:
+ *
+ * @code{c}
+ *   *ptr |= (1 << bit);
+ * @endcode
+ *
+ * There is a read-modify-write cycle occurring within the core, but this cycle
+ * is atomic and can not be disrupted by IRQs
+ *
+ * @param[in]  ptr pointer to target word
+ * @param[in]  bit bit number within the word
+ */
+static inline void bit_set32(volatile uint32_t *ptr, uint8_t bit)
```

Comment:
I'd rather rename the header bit.h if it is necessary, since bit banding is only one of the implementations